torch/testing/_internal/common_optimizers.py

```diff
@@ -183,6 +184,52 @@ def test_wrapper(*args, **kwargs):
                 raise ex
 
 
+# ------------------------------------------------------------------------------------------
+# NOTE: [optimizer kwarg categories]
+# We categorize optimizer kwargs as 3 types:
+#  1. optimizer-specific flags are like amsgrad or rho or beta, flags that are specific to
+#     algorithms and thus only show up for certain optimizers. There are many of these, so I
+#     do not bother gathering them all and listing them here. The converse to these would be
+#     global flags that every optimizer ideally _should_ support. We break global flags into
+#     2 further categories and list them all below.
+#  2. global-friendly = ["lr", "weight_decay", "maximize", "capturable"]
+#     global-friendly flags are global flags who play nicely with all other global flags,
+#     i.e., are mutually exclusive in function. This means that any pair of the following
```

Comment:
Feels like this i.e. says the opposite of the rest of this paragraph?

Reply:
mutually exclusive in function doesn’t mean exclusive in the social sense tho